public/app/plugins/datasource/opentsdb/datasource.ts

```diff
@@ -84,6 +84,10 @@ export default class OpenTsDatasource extends DataSourceApi<OpenTsdbQuery, OpenT
     });
 
     return this.performTimeSeriesQuery(queries, start, end).pipe(
+      catchError((err) => {
+        // Throw the error message here instead of the whole object to workaround the error parsing error.
+        throw err?.data?.error?.message || 'Error performing time series query.';
```

Comment:
Where is the error parsing code? Shouldn't it handle errors there in more robust way?

Reply:
So the error comes back in this format
```json
{
  "error": {
    "code": 400,
    "message": "No such name for 'metrics': 'a'",
    "details": "No such name for 'metrics': 'a'",
    "trace": "net.opentsdb.tsd.BadRequestException: No such name for 'metrics': 'a'\n\tat net.opentsdb.tsd.QueryRpc$1ErrorCB.call(QueryRpc.java:216) [tsdb-2.4.0.jar:]\n\tat"
  }
}
```

And the code error parsing code is here: https://github.com/grafana/grafana/blob/master/packages/grafana-runtime/src/utils/queryResponse.ts#L150

Should we make changes here instead?

Reply:
Hmm no I think it makes sense that we massage the opentsdb error a bit so it is normalized for the query pipeline. What seems of to me is that in the original issue, before this fix, it resulted in a react error that would remove the whole page. From the code in https://github.com/grafana/grafana/blob/master/packages/grafana-runtime/src/utils/queryResponse.ts#L150 I would just assume we show some non helpful message or no message at all but not that the whole explore would crash. So That what I mean whether we don't have some weird code somewhere in react that just crashes any time the error from the query is a bit different than expected.